draft-ietf-gnap-core-protocol.md

```diff
@@ -1451,30 +1453,44 @@ The client instance could also be signaled that interaction has completed by the
 callbacks. To facilitate all of these modes, the client instance declares the means that it 
 can interact using the `interact` field. 
 
-The `interact` field is a JSON object with keys that declare
-different interaction modes. A client instance MUST NOT declare an
-interaction mode it does not support.
+The `interact` field is a JSON object with three keys whose values declare how the client can initiate
+and complete the request, as well as provide hints to the AS about user preferences such as locale. 
+A client instance MUST NOT declare an interaction mode it does not support.
 The client instance MAY send multiple modes in the same request.
 There is no preference order specified in this request. An AS MAY
 [respond to any, all, or none of the presented interaction modes](#response-interact) in a request, depending on
-its capabilities and what is allowed to fulfill the request. This specification
-defines the following interaction modes:
+its capabilities and what is allowed to fulfill the request. 
 
-redirect (boolean)
+The `interact` field MUST contain the `start` key, and MAY contain the `finish` and `hints` keys. The value
+of each key is an array which contains strings or JSON objects as defined below.
+
+#### Start Mode Definitions
+
+This specification defines the following interaction start modes as string values under the `start` key:
+
+"redirect"
 : Indicates that the client instance can direct the end-user to an arbitrary URL
     at the AS for interaction. {{request-interact-redirect}}
 
-app (boolean)
+"app"
 : Indicates that the client instance can launch an application on the end-user's
     device for interaction. {{request-interact-app}}
 
+"user_code"
+: Indicates that the client instance can communicate a human-readable short
+    code to the end-user for use with a stable URL at the AS. {{request-interact-usercode}}
+
+#### Finish Mode Definitions
+
+This specification defines the following interaction completion modes as objects under the `finish` key:
```

Comment:
No, they are members of the array that's under the `finish` key. I must say I am confused by this structure.

Reply:
My comment may be moot if you change the format in line with #122.

Reply:
Thanks, yep this has been fixed based on the discussion in #122